api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -217,17 +219,50 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 				toAdd.add(newObs);
 			}
 		}
-		
+
+		// remove all obs in toRemove recursively from encounter
+		Set<Obs> seenIt = new LinkedHashSet<>();
```

Comment:
If you are not sure whether it will happen or not, then for now, you can start simple by assuming it will not happen. 😊